### ISSUE TITLE
http port not configurable 

### ISSUE THREAD

**Issue by johntdyer** [allenporter/stream-addons#23]
It appears this is hard coded and I have a port conflict I would like to override

```bash
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[13:42:10] INFO: Starting discovery service
[13:42:10] INFO: Starting RSTPtoWebRTC
2022/02/03 13:42:10 Server Start Awaiting Signal
2022/02/03 13:42:10 Start HTTP Server error listen tcp :8083: bind: address already in use
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

**Comment by alexyao2015** [comment]
@allenporter It seems that this uses host networking so it doesn't respect Docker port mappings, in fact, the port mappings do nothing atm. I'm not familiar with how the RTSPtoWeb server works, but does it require host networking? (i.e. does it use multicast?) If it doesn't, its best to disable it, otherwise, it would be best to provide options to configure the ports in the config using a startup script.

**Comment by allenporter** [comment]
@johntdyer thanks for the bug report, agree it makes sense to expose as configurable.

@alexyao2015 sure, this addon is just using the port from upstream default config https://github.com/deepch/RTSPtoWeb/blob/master/config.json

I imagine changing the default would break anyone who already installed the integration, hrm.

**Comment by allenporter** [comment]
I'm playing around with this in https://github.com/allenporter/stream-addons/pull/24

One thing i'm not sure how to do is pull the webui port link from an option.  It seems like using ports from options maybe wasn't intended.

**Comment by alexyao2015** [comment]
Yeah I don't believe there is a way to do that normally, however, that config really only serves to provide a clickable link since it's running in host mode. If you really want to set it though, I think the supervisor API is an option https://developers.home-assistant.io/docs/api/supervisor/endpoints/

**Comment by allenporter** [comment]
Nice, which of those supervisor APIs can set this? I wasn't quite able to find it.

**Comment by alexyao2015** [comment]
Seems like `/addons/<addon>/options` should be able to do it. However, using the API doesn't seem that great as it requires requesting a token, increasing the addon privileges, for effectively a decorative change.

**Comment by felipecrs** [comment]
Why this addon uses:

https://github.com/allenporter/stream-addons/blob/50f101d84d9ce8b35a3387304e83deae4967c2a4/rtsp-to-web/config.yaml#L15

?

**Comment by DIYtechie** [comment]
Having the same issue. log reports port conflict. But I'm not aware of any other containers using port 8083 in my setup. Hopefully a way to change it would resolve the issue nonetheless.

**Comment by ShlomiPorush** [comment]
same here. Even trying to change the port from 8083 per other port... saving the settings and restarting the addon and it does not take effect - he is still trying to map it to Port 8083...

time="2022-02-14T21:43:34+02:00" level=info msg="Server CORE start" func=main module=main
time="2022-02-14T21:43:34+02:00" level=info msg="Server HTTP start" call=Start func=RTSPServer module=http_server
time="2022-02-14T21:43:34+02:00" level=info msg="Server RTSP start" call=Start func=RTSPServer module=rtsp_server
time="2022-02-14T21:43:34+02:00" level=info msg="Server start success a wait signals" func=main module=main
time="2022-02-14T21:43:34+02:00" level=fatal msg="listen tcp :8083: bind: address already in use" call=ServerHTTPPort func=HTTPAPIServer module=http_router
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and **exiting.**

![1](https://user-images.githubusercontent.com/5992161/153934974-111b4747-8179-4f23-a60c-12c8d22dc0bb.png)

**Comment by ronsum232** [comment]
If you're using the custom WebRTC Camera add-on, that is what is using port 8083.

**Comment by ShlomiPorush** [comment]
You mean, the WebRTC add on from HACS? Yes I'm using it.. Don't want to uninstall it before this one works.

**Comment by felipecrs** [comment]
On my tests, both cannot work in parallel.

**Comment by allenporter** [comment]
An attempt to fix this will be in 1.1.0
